### ISSUE TITLE
[ic-footer] Accessibility violations on footer with grouped links

### ISSUE THREAD

**Issue by MI6-286** [mi6/ic-ui-kit#2314]
## Summary of the bug 
When viewing a footer with grouped links on a small screen, there are multiple accessibility violations.

## 🪜 How to reproduce 
1. Go to a footer with grouped links, e.g. [Grouped links story](https://mi6.github.io/ic-ui-kit/branches/develop/web-components/?path=/story/web-components-footer--grouped-links)
2. Make the screen / window size small enough to make the groups of links collapse
3. Expand one of the link groups
4. Open the 'Accessibility' tab
5. See that there are three accessibility violations listed

## 📸 Screenshots or code 
![image](https://github.com/user-attachments/assets/add962f1-a6dc-4707-b59a-6a917a37ec95)

![image](https://github.com/user-attachments/assets/b0097e8c-8bf1-4348-bdde-7f3ebbb41d83)

## Additional info 
Uncomment cy.checkA11yWithWait() on relevant tests within IcFooter Cypress tests.

**Comment by gd2910** [comment]
Fixed in https://github.com/mi6/ic-ui-kit/pull/2790
